pkg/serviceaccount/metadata/metadata.go

```diff
@@ -0,0 +1,159 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package metadata is a separate subpackage, because many controllers
+// also import pkg/serviceaccount, but are not allowed by import-boss
+// to depend on go-restful. This way, this code stays close to other related
+// code, but doesn't add go-restful as a dependency of those controllers.
+package metadata
+
+import (
+	"encoding/json"
+	"net/http"
+	"sort"
+
+	restful "github.com/emicklei/go-restful"
+	jose "gopkg.in/square/go-jose.v2"
+
+	"k8s.io/klog"
+	"k8s.io/kubernetes/pkg/serviceaccount"
+)
+
+const (
+	// OpenIDConfigPath is the URL path at which the API server serves
+	// an OIDC Provider Configuration Information document, corresponding
+	// to the Kubernetes Service Account key issuer.
+	// https://openid.net/specs/openid-connect-discovery-1_0.html
+	OpenIDConfigPath = "/.well-known/openid-configuration"
+	// JWKSPath is the URL path at which the API server serves a JWKS
+	// containing the public keys that may be used to sign Kubernetes
+	// Service Account keys
+	JWKSPath = "/jwks"
```

Comment:
I wonder if there's a better path for this.

```
/serviceaccounts/keys
/serviceaccounts/v1alpha1/keys
```

@liggitt wdyt?

Reply:
LOL we had this argument in above comments ;).

Reply:
I'm open to whatever.